src/ast/semantic_analyser.cpp

```diff
@@ -478,14 +478,54 @@ void SemanticAnalyser::visit(Call &call)
     }
     call.type = SizedType(Type::integer, 8);
   }
-   else if (call.func == "uaddr")
-   {
-    if (check_nargs(call, 1)) {
-      if (check_arg(call, Type::string, 0, true)) {
-        check_symbol(call, 0);
+  else if (call.func == "uaddr")
+  {
+    if (! check_nargs(call, 1)) return;
+    if (! (check_arg(call, Type::string, 0, true) && check_symbol(call, 0)))
+      return;
+
+    std::vector<int> sizes;
+    auto &name = static_cast<String&>(*call.vargs->at(0)).str;
+    for (auto &ap : *probe_->attach_points) {
+      ProbeType type = probetype(ap->provider);
+      if (type != ProbeType::usdt && type != ProbeType::uretprobe &&
+          type != ProbeType::uprobe) {
+        bpftrace_.error(err_, call.loc,
+            "uaddr can only be used with u(ret)probes and usdt probes");
+        sizes.push_back(0);
+        continue;
+      }
+      struct symbol sym = {};
+      int err = bpftrace_.resolve_uname(name, &sym, ap->target);
+      if (err < 0 || sym.address == 0) {
+        bpftrace_.error(err_, call.loc,
+                        "Could not resolve symbol: " + ap->target + ":" + name);
+      }
+      sizes.push_back(sym.size);
+    }
+
+    for (size_t i = 0; i < sizes.size(); i++) {
```

Comment:
just a minor comment...
```suggestion
    for (size_t i = 1; i < sizes.size(); i++) {
```